CMakeLists.txt

```diff
@@ -235,7 +235,15 @@ file(GLOB SOURCES
 )
 
 add_executable(lightgbm src/main.cpp ${SOURCES})
-add_library(_lightgbm SHARED src/c_api.cpp src/lightgbm_R.cpp ${SOURCES})
+set(SOURCES "${SOURCES} src/c_api.cpp")
```

Comment:
I think `append` should fix CI failures
https://cmake.org/cmake/help/v3.12/command/list.html#append

Reply:
ah ok, I shouldn't have assumed it was just a space-delimited list. Let me try that

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/2911/commits/8d5b652c43383e66ca0710498c10eb5b915f9180